engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsValueOfTypeNode.java

```diff
@@ -31,12 +31,6 @@ public static IsValueOfTypeNode build() {
 
   public abstract boolean execute(Object expectedType, Object payload);
 
-  @Specialization
-  boolean doAtom(AtomConstructor consr, Object payload) {
-    Object expectedType = consr.getType();
-    return execute(expectedType, payload);
```

Comment:
I cannot add a comment to the fallback.

What do you think about throwing Illegal Argument there instead of returning false?

Imo if I call is_a with atom constructor, which will now **never** be true , that is a code error and potential bug and I should be notified.

Same goes for `x.is_a 42`  - it does not make sense to make such a call so I think we should error on it.

It would probably require to slightly amend how we handle the true cases. Or just add a check if the type argument really is either an Enso type or a Java class. I think it would be helpful to catch erroneous usages.